### PR TITLE
TRUNK-5213: @OpenmrsProfile loads Spring resources conditionally based on missing module(s)

### PR DESCRIPTION
**Pull request by mks-d**
## Description
`@OpenmrsProfile` loads Spring resources conditionally based on missing module(s)
## Related Issue

https://issues.openmrs.org/browse/TRUNK-5213

## Checklist:

- [x] My pull request only contains one single commit.
- [x] My pull request is based on the latest master branch
  `git pull --rebase upstream master`.
- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by mks-d** [comment]
@dkayiwa , let me know if you're ok with this now that TRUNK-5216 is solved.

File: api/src/test/java/org/openmrs/annotation/OpenmrsProfileWithoutTest1Module.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.annotation;
+
+/**
+ * Test bean which should be only loaded when htmlformentry module is missing
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/annotation/OpenmrsProfileWithoutTest1Module.java]: When which module is missing? htmlformentry or test1?

File: api/src/main/java/org/openmrs/annotation/OpenmrsProfileExcludeFilter.java [hunk 2]
```diff
@@ -40,48 +40,54 @@
...
-			if (!moduleMatched) {
-				return false;
+			else {
+				String[] splitModuleAndVersion = moduleAndVersion.split(":");
+				String moduleId = splitModuleAndVersion[0];
+				String moduleVersion = splitModuleAndVersion[1];
+
+				boolean moduleMatched = false;
+				for (Module module : ModuleFactory.getStartedModules()) {
+					if (module.getModuleId().equals(moduleId) && ModuleUtil.matchRequiredVersions(module.getVersion(), moduleVersion)) {
```
[the remark this answers is not in the document]
Comment (dkayiwa) [reply api/src/main/java/org/openmrs/annotation/OpenmrsProfileExcludeFilter.java]: Did you see the above comment from codacy? 😊

[the remark this answers is not in the document]
Comment (mks-d) [reply api/src/main/java/org/openmrs/annotation/OpenmrsProfileExcludeFilter.java]: I don't get that, is the formatter not taking care of this?

**Pull request merged**